app/models/property.rb

```diff
@@ -33,6 +33,7 @@ class Property < ApplicationRecord
   belongs_to :user
   has_many :property_advertisers
   has_many :advertisers, through: :property_advertisers, class_name: "User", foreign_key: "advertiser_id"
+  has_many :impressions
```

Comment:
I intentionally left this relationship out in favor of the method below to improve query performance. I can rethink the `impressions` method into how we can move it into a scope if needed. Query performance without joining on `property_advertisers` will be problematic since all partition tables will need to be scanned.